Implementación de selector de cantidad

evertms opened the pull request:
Se implementó un componente reutilizable QuantitySelector para la selección de cantidades de productos con las siguientes características:
- Validación de límites (1-99)
- Manejo de errores en tiempo real
- Prevención de valores inválidos
- Botones + y - para ajuste rápido
- Campo de entrada numérica con validación
- Integración con el modal de productos
- Corrección del comportamiento del botón Aceptar
- Mejora en la experiencia de usuario al iniciar con campo vacío Los cambios incluyen validaciones robustas para asegurar que solo se puedan agregar productos con cantidades válidas, mostrando mensajes de error apropiados y deshabilitando el botón de Aceptar cuando sea necesario.

evertms committed:
Implementación de selector de cantidad

Se implementó un componente reutilizable QuantitySelector para la selección de cantidades de productos con las siguientes características:
- Validación de límites (1-99)
- Manejo de errores en tiempo real
- Prevención de valores inválidos
- Botones + y - para ajuste rápido
- Campo de entrada numérica con validación
- Integración con el modal de productos
- Corrección del comportamiento del botón Aceptar
- Mejora en la experiencia de usuario al iniciar con campo vacío
Los cambios incluyen validaciones robustas para asegurar que solo se puedan agregar productos con cantidades válidas, mostrando mensajes de error apropiados y deshabilitando el botón de Aceptar cuando sea necesario.

evertms commented:
@RaidenMakio @Matipala

RaidenMT commented:
Adjuntar link de HU y copiar y pegar descripción

evertms commented:
**Descripción:** Como usuario, quiero seleccionar la cantidad de unidades de un producto que deseo agregar al pedido, ya sea escribiendo el número directamente o ajustando la cantidad con botones `+` y `-`, para tener flexibilidad y control sobre mi elección.\
**Criterios de aceptación:**

1. **Selección mediante botones** `+` **y** `-`**:**
   1. Dado que estoy en la página de un producto,\
      Cuando presiono el botón `+`,\
      Entonces la cantidad aumenta en 1 unidad hasta un máximo de 99.
   2. Dado que estoy en la página de un producto,\
      Cuando presiono el botón `-`,\
      Entonces la cantidad disminuye en 1 unidad hasta un mínimo de 1.
   3. Dado que la cantidad actual es 1,\
      Cuando presiono el botón `-`,\
      Entonces la cantidad no disminuye ya que el botón `-` se desactiva y no permite reducir más la cantidad.
   4. Dado que la cantidad actual es 99,\
      Cuando presiono el botón `+`,\
      Entonces la cantidad no aumenta ya que el botón `+` se desactiva y no permite aumentar más la cantidad.
2. **Entrada directa mediante un campo numérico:**
   1. Dado que estoy en la página de un producto,\
      Cuando ingreso un número en el campo de cantidad (entre 1 y 99),\
      Entonces el valor se actualiza correctamente.
   2. Dado que estoy en la página de un producto,\
      Cuando intento ingresar un número menor a 1 o mayor a 99,\
      Entonces el sistema no permite la entrada y muestra un mensaje de error: "La cantidad debe estar entre 1 y 99".
   3. Dado que estoy en la página de un producto,\
      Cuando intento ingresar texto o caracteres no numéricos en el campo de cantidad,\
      Entonces el sistema no permite la entrada y muestra un mensaje de error: "Solo se permiten números".

http://159.69.123.44/isw/browse/TALLE-249/

RaidenMakio commented on OrderNow-React/src/components/ProductDetailsModal.jsx:
```diff
@@ -1,18 +1,20 @@
+  const handleAccept = (e) => {
+    e.stopPropagation();
+    if (!hasError && !isNaN(parseInt(quantity, 10))) {
+      console.log("Producto seleccionado:", product, "Cantidad:", quantity);
```
Borrar console.log

RaidenMakio commented on OrderNow-React/src/components/ProductDetailsModal.jsx:
```diff
@@ -21,12 +23,18 @@ const ProductDetailsModal = ({ product, closeModal }) => {
+        className="relative bg-white rounded-lg shadow-lg z-10 max-w-md w-full mx-4 p-6"
+        onClick={handleModalClick}
+      >
         <button
```
Utilizar componente Button

RaidenMakio commented on OrderNow-React/src/components/ProductDetailsModal.jsx:
```diff
@@ -40,30 +48,22 @@ const ProductDetailsModal = ({ product, closeModal }) => {
         </div>
+        
         <div className="flex justify-center">
           <button
```
Componente Button

RaidenMakio commented on OrderNow-React/src/components/atoms/QuantitySelector.jsx:
```diff
@@ -0,0 +1,139 @@
+    if (quantity === '') {
+      setQuantity(1);
+      validateQuantity(1);
+    } else if (quantity < maxQuantity) {
```
Quitar indentacion de else

RaidenMakio commented on OrderNow-React/src/components/atoms/QuantitySelector.jsx:
```diff
@@ -0,0 +1,139 @@
+  };
+
+  const decrement = () => {
+    if (quantity > minQuantity) {
```
Invertir if y hacer return

RaidenMakio commented on OrderNow-React/src/components/atoms/QuantitySelector.jsx:
```diff
@@ -0,0 +1,139 @@
+    
+    if (isNaN(numValue) || numValue < minQuantity || numValue > maxQuantity) {
+      validateQuantity(quantity);
+    } else {
```
Quitar indentacion else

RaidenMakio commented on OrderNow-React/src/components/atoms/QuantitySelector.jsx:
```diff
@@ -0,0 +1,139 @@
+    <div className="flex flex-col items-center">
+      <div className={`py-2 px-3 inline-block bg-white border rounded-lg ${error ? 'border-red-500' : 'border-gray-200'}`}>
+        <div className="flex items-center gap-x-1.5">
+          <button 
```
Componente Button

RaidenMakio reviewed (changes requested):
Corregir

evertms committed:
Corrección de las observaciones

Se mejoraron las prácticas con los condicionales según las observaciones del docente. Se utilizó el componente Button y fue modificado para cumplir con los criterios de aceptación.

evertms committed:
Corrección de conflictos y errores

Unifiqué los dos componentes Button, se quedó en la ruta src/components/atoms. Eliminé el button.stories (ya hablé con Daniel al respecto). Lo demás son correcciones de conflictos con ambos archivos. Ahora todo funciona correctamente y reutilizando componentes.

evertms committed:
Uso de Button en '+' y '-'

Se realizaron los siguientes cambios:
1. En Button.jsx:
- Agregada variante "quantity" para botones de incremento/decremento
- Añadido soporte para íconos SVG mediante prop icon
- Implementados estilos específicos para botones de cantidad
- Mejorado manejo de estados deshabilitados
2. En QuantitySelector.jsx:
- Reemplazados botones nativos por componente Button con variante "quantity"
- Integrados íconos SVG a través del prop icon
- Mantenida funcionalidad original con mejor reutilización de componentes.
Los cambios mejoran la consistencia del diseño y la reutilización de componentes manteniendo toda la funcionalidad existente.

evertms committed:
Merge remote-tracking branch 'upstream/main'

evertms committed:
Merge remote-tracking branch 'upstream/main'

evertms committed:
Merge remote-tracking branch 'upstream/main'
